components/movementsensor/adxl345/adxl345.go

```diff
@@ -291,10 +507,33 @@ func (adxl *adxl345) Accuracy(ctx context.Context, extra map[string]interface{})
 	return map[string]float32{}, movementsensor.ErrMethodUnimplementedAccuracy
 }
 
+func contains(s []string, str string) bool {
+	for _, v := range s {
+		if v == str {
+			return true
+		}
+	}
+	return false
+}
+
 func (adxl *adxl345) Readings(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error) {
 	adxl.mu.Lock()
 	defer adxl.mu.Unlock()
-	return map[string]interface{}{"linear_acceleration": adxl.linearAcceleration}, adxl.err.Get()
+
+	readings := make(map[string]interface{})
+	readings["linear_acceleration"] = adxl.linearAcceleration
+	readings["data_ready"] = contains(adxl.interruptsFound, DataReady)
+	readings["single_tap"] = contains(adxl.interruptsFound, SingleTap)
+	readings["double_tap"] = contains(adxl.interruptsFound, DoubleTap)
+	readings["activity"] = contains(adxl.interruptsFound, Activity)
+	readings["inactivity"] = contains(adxl.interruptsFound, Inactivity)
+	readings["freefall"] = contains(adxl.interruptsFound, Freefall)
+	readings["watermark"] = contains(adxl.interruptsFound, Watermark)
+	readings["overrun"] = contains(adxl.interruptsFound, Overrun)
```

Comment:
This whole approach of parsing the byte into a list of strings and then parsing the list into 8 bits feels very inefficient to me. If this is all we send back to the user, how about just storing the byte read from the register, and using bitmasks to populate these fields?

```go
func (adxl *adxl345) readInterrupts(ctx context.Context) {
    intSourceRegister, err := adxl.readByte(ctx, IntSource)
    if err != nil {
        return err
    }
    adxl.intData = intSourceRegister
}
...
func (adxl *adxl345) Readings(...) {
    ...
    readings["inactivity"] = (adxl.intData & adxl.interruptsEnabled & (1 << 3)) != 0
    readings["activity"]   = (adxl.intData & adxl.interruptsEnabled & (1 << 4)) != 0
    readings["overrun"]    = (adxl.intData & adxl.interruptsEnabled & (1 << 0)) != 0
    ...
    adxl.intData = 0
}
```

Reply:
...and now that I re-read the version I wrote, it's probably more like
```
adxl.intData = intSourceRegister & adxl.interruptsEnabled
...
readings["..."] = (adxl.intData & (1 << n)) != 0
```